update-agent/dbus/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+//! Query the current update status:
+//! ```bash
+//! gdbus call --session -d org.worldcoin.UpdateProgress1 -o '/org/worldcoin/UpdateProgress1' -m \
+//! org.freedesktop.DBus.Properties.Get org.worldcoin.UpdateProgress1 Status
+//! ```
+//!
+//! Montor for signals:
+//! ```bash
+//! dbus-monitor type='signal',sender='org.worldcoin.UpdateProgress1'
+//! ```
+
+use serde::{Deserialize, Serialize};
+use zbus::interface;
+use zbus::zvariant::{OwnedValue, Type, Value};
+
+/// A trait representing update progress behavior.
+///
+/// This trait is implemented by types that can provide information about the current update status.
+pub trait UpdateProgressT: Send + Sync + 'static {
+    fn status(&self) -> UpdateStatus;
+}
+
+/// A wrapper struct for types implementing [`UpdateProgressT`].
+pub struct UpdateProgress<T: UpdateProgressT>(pub T);
+
+#[derive(Debug, Serialize, Deserialize, Type, Clone, Value, OwnedValue)]
```

Comment:
Might as well make it `Copy`
```suggestion
#[derive(Debug, Serialize, Deserialize, Type, Clone, Copy, Value, OwnedValue)]
```